main.tf

```diff
@@ -1,3 +1,65 @@
+# Required for public ECR where Karpenter artifacts are hosted
+provider "aws" {
+  region = "us-west-2"
+  alias  = "oregon"
+}
+
+provider "kubernetes" {
+  host                   = module.eks_blueprints.eks_cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+  token                  = data.aws_eks_cluster_auth.this.token
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.eks_blueprints.eks_cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+    token                  = data.aws_eks_cluster_auth.this.token
+  }
+}
+
+provider "kubectl" {
+  apply_retry_count      = 10
+  host                   = module.eks_blueprints.eks_cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+  load_config_file       = false
+  token                  = data.aws_eks_cluster_auth.this.token
+}
+
+module "eks_blueprints" {
+
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.27.0"
+  cluster_name    = local.name
+
+  # EKS Cluster VPC and Subnet mandatory config
+  vpc_id             = module.vpc.vpc_id
+  private_subnet_ids = module.vpc.private_subnets
+
+  # EKS CONTROL PLANE VARIABLES
+  cluster_version = local.cluster_version
+
+  # List of Additional roles admin in the cluster
```

Comment:
Read all these comments. It's very helpful.

Reply:
ok, sure